src/Conversion/ONNXToTOSA/NN/MaxPoolSingleOut.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- MaxPoolSingleOut.cpp - MaxPoolSingleOut Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX MaxpoolSingleOut operator to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "mlir/IR/Value.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "llvm/ADT/ArrayRef.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXMaxPoolSingleOutOpLoweringToTOSA : public OpConversionPattern<ONNXMaxPoolSingleOutOp> {
+public:
+  using OpConversionPattern<ONNXMaxPoolSingleOutOp>::OpConversionPattern;
+  using OpAdaptor = typename ONNXMaxPoolSingleOutOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXMaxPoolSingleOutOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+
+    Location loc = op->getLoc();
+
+    Value Input = adaptor.X();
+    // - The attribute Ceil_mode is ignored because its effect is already impacting the return
+    // type of the operator
+    // - The attribute auto_pad is ignored because it is marked as deprecated for ONNX
+    // - The attributes storage_order and dilations is also unsupported
+    mlir::StringAttr autoPad = adaptor.auto_padAttr();
+    
+    mlir::ArrayAttr dilations = adaptor.dilationsAttr();
+    mlir::ArrayAttr kernelShape = adaptor.kernel_shapeAttr();
+    mlir::ArrayAttr pads = adaptor.padsAttr();
+    mlir::ArrayAttr strides = adaptor.stridesAttr();
+
+    auto InputType = Input.getType().cast<RankedTensorType>();
+    auto resultType = op.getResult().getType();
+    auto resultShape = resultType.cast<RankedTensorType>().getShape();
+    // Construct the transposed type for the new MaxPool OP
+    Type newResultType = RankedTensorType::get(
+      {resultShape[0], resultShape[2], resultShape[3], resultShape[1]}, InputType.getElementType());
+
+    if (autoPad != "NOTSET") {
```

Comment:
MLIR has a different mechanism for these type of print statements. See [MLIR doc](https://mlir.llvm.org/getting_started/Debugging/#:~:text=Miscellaneous%20tips%C2%A0,on%2Ddiagnostic.). Aside, I think the conversion should better fail if those are set.

Reply:
Good catch, I didn't know about it